### PR TITLE
Do not show deactivated users in sharees and contacts

### LINKED ISSUE #5608: Deactivated Accounts still show up in Contacts for Chat and Videocalls
Opened here as per receommendation of [this closed issue](https://github.com/nextcloud/contacts/issues/238).

### Steps to reproduce
Deactivate an account

### Expected behaviour
Accounts should be unavailable for chat and videocalls

### Actual behaviour
Contacts show up as able to chat and for videocalls

### Server configuration

**Nextcloud version:** (see Nextcloud admin page)
12.0.0
**Contacts version:** (see Nextcloud apps page)
1.5.3
**Updated from an older Nextcloud or fresh install:**
Updated since owncloud 6, migrated to Nextcloud 10 from owncloud 9

### PR DESCRIPTION
**Pull request by tcitworld**
First part to fix #5608 : remove deactivated users from system address book.

### CODE REVIEW

**Comment by tcitworld** [comment]
The second part of fixes lacks tests into `Test\Collaboration\Collaborators\UserPluginTest`. I didn't manage to get that part in time, if someone wants to have a quick look.

**Comment by skjnldsv** [comment]
@tcitworld ready for reviews?

**Comment by tcitworld** [comment]
@skjnldsv Yes, it just lacks one more test.

**Review by skjnldsv — APPROVED** [review]
Code is ok :+1:

File: apps/dav/tests/unit/CardDAV/SyncServiceTest.php [hunk 1]
```diff
@@ -106,6 +106,7 @@ public function testUpdateAndDeleteUser() {
 		$user->method('getUID')->willReturn('test-user');
 		$user->method('getCloudId')->willReturn('cloudId');
 		$user->method('getDisplayName')->willReturn('test-user');
+		$user->method('isEnabled')->willReturn(true);
```
Comment (ChristophWurst) [inline apps/dav/tests/unit/CardDAV/SyncServiceTest.php]: would be great to have a test for the `false` condition as well :wink:

Comment (tcitworld) [reply apps/dav/tests/unit/CardDAV/SyncServiceTest.php]: @ChristophWurst How's that ?

Comment (ChristophWurst) [reply apps/dav/tests/unit/CardDAV/SyncServiceTest.php]: I think you just pushed what I was asking for :wink:

**Comment by rullzer** [comment]
@ChristophWurst so good to go from you?

**Review by rullzer — APPROVED** [review]
Makes sense :rocket:

**Pull request merged**